pkg/hubble/relay/observer/server.go

```diff
@@ -168,6 +169,65 @@ sortedFlowsLoop:
 	return g.Wait()
 }
 
+// GetNodes implements observerpb.ObserverClient.GetNodes.
+func (s *Server) GetNodes(ctx context.Context, req *observerpb.GetNodesRequest) (*observerpb.GetNodesResponse, error) {
+	if md, ok := metadata.FromIncomingContext(ctx); ok {
+		ctx = metadata.NewOutgoingContext(ctx, md)
+	}
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	g, ctx := errgroup.WithContext(ctx)
+
+	peers := s.peers.List()
+	nodes := make([]*observerpb.Node, 0, len(peers))
+	for _, p := range peers {
+		p := p
```

Comment:
Nit: should this be moved right before the call to `g.Go()`? It's not strictly necessary to shadow at the start of the loop, having it closer to the closure's definition better conveys why it's necessary. Also makes it less likely to be overlooked in the future. :wink: